src/Chapter4.hs

```diff
@@ -673,6 +707,23 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a
+    = EmptyNode
+    | Node a (Tree a) (Tree a)
+
+instance Functor Tree where
+    fmap :: (a -> b) -> Tree a -> Tree b
+    fmap _ EmptyNode      = EmptyNode
+    fmap f (Node x lx rx) = Node (f x) (fmap f lx) (fmap f rx)
+
+reverseTree :: Tree a -> Tree a
+reverseTree EmptyNode      = EmptyNode
+reverseTree (Node x lx rx) = Node x (reverseTree lx) (reverseTree rx)
```

Comment:
In this case it won't be change, unfortunately. You will just walk through the Tree.

You will need to do the following:

```suggestion
reverseTree (Node x lx rx) = Node x (reverseTree rx) (reverseTree lx)
```